File: app/src/main/java/com/app/gitrepository/data/DataManager.java
package com.app.gitrepository.data;

import com.hkuaapp.data.local.db.DbHelper;
import com.hkuaapp.data.local.prefs.PreferenceHelper;
import com.app.gitrepository.data.remote.ApiHelper;

public interface DataManager extends PreferenceHelper, ApiHelper, DbHelper {

    void setUserAsLoggedOut();

    void updateApiHeader(Long userId, String accessToken);

}

File: app/src/main/java/com/app/gitrepository/data/DataManager.kt
package com.app.gitrepository.data

import com.app.gitrepository.data.local.DbHelper
import com.app.gitrepository.data.model.Repository
import com.app.gitrepository.data.network.ApiHelper
import io.reactivex.Single

interface DataManager : ApiHelper,DbHelper{
    fun isNetworkConnected() : Boolean
    fun startWorker()
}